### PR TITLE
Remove GPU dependency from ProfileObserver

### PR DESCRIPTION
**Pull request by mdschatz**
Summary: Remove GPU dependency and register ProfileObserver.

Differential Revision: D14265801

### CODE REVIEW

**Comment by ezyang** [comment]
This isn't just removing the dependency, this is removing CUDA functionality completely! Maybe that's the right thing to do (I am not familiar with ProfileObserver), but let the commit message reflect the contents of the diff.

**Comment by ezyang** [comment]
It looks like you're the one who added this in the first place, so I'll LGTM it :)

**Review by ezyang — APPROVED** [review]

**Comment by mdschatz** [comment]
Yep, I'm fine removing CUDA functionality.  It's creating some headaches and we've only really used this for CPU profiling anyway.

File: caffe2/observers/CMakeLists.txt [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
   message(STATUS "Include Observer library")
-  set(GLOB profile_observer_files profile_observer_*.cc)
   set(Caffe2_CONTRIB_OBSERVERS_CPU_SRC
+    "${CMAKE_CURRENT_SOURCE_DIR}/profile_observer.cc"
     "${CMAKE_CURRENT_SOURCE_DIR}/time_observer.cc"
     "${CMAKE_CURRENT_SOURCE_DIR}/runcnt_observer.cc"
   )
-  set(Caffe2_CONTRIB_OBSERVERS_GPU_SRC
-    "${CMAKE_CURRENT_SOURCE_DIR}/profile_observer_gpu.cc"
-  )
```
Comment (pjh5) [inline caffe2/observers/CMakeLists.txt]: You can delete https://github.com/pytorch/pytorch/blob/master/caffe2/observers/CMakeLists.txt#L15-L16 as well

**Review by pjh5 — COMMENTED** [review]
I think you need to add CAFFE2_API to https://github.com/pytorch/pytorch/blob/master/caffe2/observers/profile_observer.h#L99 just like https://github.com/pytorch/pytorch/blob/master/caffe2/observers/time_observer.h#L48 and also to ProfileOperatorObserver etc

**Pull request closed without merging**